experimental/builder/test/conv/test_ckb_conv_fwd_2d_dl_fp16.cpp

```diff
@@ -54,4 +59,104 @@ TEST(FwdConvInstances,
         {"DeviceGroupedConvFwdDlMultipleD_NHWC_KYXC_NHWK", "256, 128, 128, 16", "Filter1x1Pad0"});
 }
 
+TEST(FwdConvInstances, Fp16_2D_DL_EndToEndBasic)
+{
+    constexpr ConvSignature signature{.spatial_dim           = 2,
+                                      .direction             = ConvDirection::FORWARD,
+                                      .layout                = GroupConvLayout2D::GNHWC_GKYXC_GNHWK,
+                                      .data_type             = DataType::FP16,
+                                      .elementwise_operation = ElementwiseOperation::PASS_THROUGH};
+
+    constexpr auto algorithm =
+        ConvAlgorithm_DeviceGroupedConvFwdDlMultipleD_NHWC_KYXC_NHWK{}
+            .with_thread_block(FwdThreadBlock_256_128x128x16)
+            .with_specializations(ConvFwdSpecialization::DEFAULT, GemmSpecialization::MNKPadding)
+            .with_dl_thread_config(DlThreadConfig_16x2x4x4x1)
+            .with_dl_thread_cluster(DlThreadCluster_8x2)
+            .with_dl_transfer(DlFwdTransfer);
+
+    auto args = ckt::ConvArgs<signature>{
+        .lengths =
+            {
+                .batch_size      = 16,
+                .groups          = 1,
+                .input_channels  = 32,
+                .output_channels = 16,
+                .image =
+                    {
+                        .width  = 56,
+                        .height = 56,
+                    },
+                .filter =
+                    {
+                        .width  = 3,
+                        .height = 3,
+                    },
+            },
+        .filter_strides  = {.width = 1, .height = 1},
+        .filter_dilation = {.width = 1, .height = 1},
+        .input_left_pad  = {.width = 0, .height = 0},
+        .input_right_pad = {.width = 0, .height = 0},
+    };
+
+    auto tmm = ckt::TensorMemoryManager<signature>(args);
```

Comment:
I think I got this incorrect in my design notes. Construction doesn't need  args, just the allocation method.